crates/metrics/src/txpool_metrics.rs

```diff
@@ -1,24 +1,94 @@
-use crate::global_registry;
-use prometheus_client::metrics::histogram::Histogram;
+use crate::{
+    buckets::{
+        buckets,
+        Buckets,
+    },
+    global_registry,
+};
+use prometheus_client::metrics::{
+    gauge::Gauge,
+    histogram::Histogram,
+};
 use std::sync::OnceLock;
 
 pub struct TxPoolMetrics {
-    pub tx_size_histogram: Histogram,
+    /// Size of transactions in the txpool in bytes
+    pub tx_size: Histogram,
+    pub number_of_transactions: Gauge,
+    pub number_of_transactions_pending_verification: Gauge,
+    pub number_of_executable_transactions: Gauge,
+    /// Time of transactions in the txpool in seconds
+    pub transaction_time_in_txpool_secs: Histogram,
+    /// Time actively spent by transaction insertion in the thread pool
+    pub transaction_insertion_time_in_thread_pool_milliseconds: Histogram,
+    /// How long it took for the selection algorithm to select transactions
+    pub select_transaction_time_nanoseconds: Histogram,
 }
 
 impl Default for TxPoolMetrics {
     fn default() -> Self {
-        let tx_sizes = Vec::new();
+        let tx_size = Histogram::new(buckets(Buckets::TransactionSize));
+        let transaction_time_in_txpool_secs = Histogram::new(buckets(Buckets::Timing));
+        let select_transaction_time_nanoseconds =
+            Histogram::new(buckets(Buckets::Timing));
```

Comment:
These buckets could be too large for a metric that is represented in nanoseconds.

Reply:
Fair. I have added a `TimingCoarseGrained` bucket that is better fit for measurements in ms and ns. 
Happy to change the bucket values if needed. (365807bd00e487835eb1a92f4c767b0659157cb1)